File: app/src/main/java/ua/hospes/rtm/core/di/module/NetModule.java
package ua.hospes.rtm.core.di.module;

import dagger.Module;
import dagger.Provides;
import ua.hospes.rtm.domain.preferences.PreferencesManager;
import ua.hospes.rtm.domain.preferences.PreferencesManagerImpl;

/**
 * @author Andrew Khloponin
 */
@Module
public class NetModule {
    @Provides
    PreferencesManager providePreferencesManager(PreferencesManagerImpl manager) {
        return manager;
    }
}

File: app/src/main/java/ua/hospes/rtm/di/module/NetModule.kt
package ua.hospes.rtm.di.module

import android.content.Context
import dagger.Module
import dagger.Provides
import ua.hospes.rtm.domain.preferences.PreferencesManager
import ua.hospes.rtm.domain.preferences.PreferencesManagerImpl
import javax.inject.Singleton

@Module
object NetModule {
    @Singleton
    @Provides @JvmStatic
    internal fun providePrefs(ctx: Context): PreferencesManager = PreferencesManagerImpl(ctx)
}